### ISSUE TITLE
[Feature Request]: Trim N characters from the beginning and/or end of every line when pasting

### ISSUE THREAD

**Issue by CharlesARoy** [sabrogden/Ditto#496]
It would be super helpful to have a special paste function that could paste copied text and trim a specified number of characters from the beginning and/or end of every line. This option would also partially address some requests for trimming whitespace on every line.

For example, say I copied some indented Python code that had 4 extra spaces in front of every line, e.g.:
```
    def my_function(input):
        print(input)
```

It would be awesome if Ditto supported trimming a constant number of characters from every line for cases in which I want to paste it elsewhere without the extra indentation, e.g.:
```
def my_function(input):
    print(input)
```

**Comment by tuwulin365** [comment]
options - advanced - on copy scripts

**Comment by CharlesARoy** [comment]
### TL:DR:  

From the [Ditto scripting documentation](https://github.com/sabrogden/Ditto/wiki/Scripting) and [this regex documentation](http://www.java2s.com/Tutorial/Java/0130__Regular-Expressions/Matchaparticularcharacteraspecifiednumberoftimes.htm) it seems like I would want to run a regex command like this:

```
clip.AsciiTextReplaceRegex(^.{4}?, "");

return false
```

My two questions about this are:
1. Is there a way to prompt the user with an input to feed to a script so that I could e.g. replace the 4 with any number?  
2. Is there a way to only run the script when I want rather than every time I copy or paste something?


------------------------------------

In my specific use case, I was copying text from Vim/Neovim running on a remote server via my terminal emulator. In doing this, I was getting both extra spaces and the additional characters in the left margin (such as the relative line numbers) which I didn't want:

![image](https://github.com/sabrogden/Ditto/assets/46253001/82ab9254-4f1e-47f9-87e2-98cb3d3a9cbd)

Copying from ^^^ pastes as:

```
     63
-       1 def main(args):
│       2 │   │
-       3 │   def start_script_logger(args):
-       4 │   │   """
```

In that case, I found that by selecting the text in visual line mode:
![image](https://github.com/sabrogden/Ditto/assets/46253001/3945d049-b251-4580-b803-18a4f91ceefe)

and then running a command like:
`:'<,'>'<,'>w ! bash -c cat | cut -c 5-`

That I could avoid copying the characters in the margin and trim any whitespace as desired.

**Comment by Chealer** [comment]
@CharlesARoy: To my (limited) knowledge, the answer to #2 is negative.

As this is an ITS, this should be retitled as an issue. I suggest one of:
1. Pasting cannot optionally trim n characters from the beginning and/or end of every line
2. Pasting cannot run custom set of On Paste scripts

**Comment by CharlesARoy** [comment]
Thanks @Chealer! I no longer use Ditto as I mostly work on Mac and Linux these days, but I updated the name of the issue as requested.
